packages/firestore/src/api/credentials.ts

```diff
@@ -29,14 +29,16 @@ import { logDebug } from '../util/log';
 // moved to an auth/ folder to match other platforms.
 
 export interface FirstPartyCredentialsSettings {
-  type: 'gapi';
-  client: unknown;
-  sessionIndex: string;
+  // These are external types. Prevent minification.
+  ['type']: 'gapi';
+  ['client']: unknown;
+  ['sessionIndex']: string;
 }
 
 export interface ProviderCredentialsSettings {
-  type: 'provider';
-  client: CredentialsProvider;
+  // These are external types. Prevent minification.
+  ['type']: 'provider';
+  ['client']: CredentialsProvider;
```

Comment:
Does it make any difference? These are just types which won't be minified anyway.

Reply:
If you are making the change to be able to do `credentials['type']`, it's not necessary. Typescript will allow you to do indexing as expected.

Reply:
This blocks minification.

Reply:
I see. that makes sense